pennylane/operation.py

```diff
@@ -129,6 +129,99 @@
 
 from .utils import pauli_eigs
 
+
+def expand_matrix(base_matrix, wires, wire_order):
```

Comment:
Josh and I thought of making this a transform, but I want it to be here in order to adapt it to future needs

Reply:
Oh, interesting, I wasn't aware of this. So we have this logic in several places! A good reason to focus it somewhere...

The code seems to be slightly different, let me benchmark it!

Reply:
> The code seems to be slightly different, let me benchmark it!

Yep, I wonder if the `utils` version might be more battle-hardened?

Reply:
`expand` seems to be a bit faster, but the torch differentiation test fails (and the utils method does not claim to be differentiable). Still it would be neat to rely on existing functions...Maybe I can figure out what is going wrong

Reply:
Ok, for now I added a note that the two should be merged some day.

Reply:
We should also open an issue! this could make a very good tech assessment

Reply:
let me know if you don't have time, and I'll will add it to my todo list tomorrow :)